test-packages/my-app-test.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+const fs = require('fs-extra');
+const util = require('util');
+const rimraf = util.promisify(require('rimraf'));
+const chai = require('chai');
+const expect = chai.expect;
+const chaiFiles = require('chai-files');
+const dir = chaiFiles.dir;
+const file = chaiFiles.file;
+const path = require('path');
+const execa = require('execa');
+
+chai.use(chaiFiles);
+
+const BASE_PATH = path.join(__dirname, 'my-app');
+
+describe('app coverage generation', function () {
+  this.timeout(10000000);
+
+  beforeEach(async function () {
+    await rimraf(`${BASE_PATH}/coverage*`);
+    await execa('git', ['clean', '-f', 'my-app'], { cwd: __dirname });
+    await execa('git', ['restore', 'my-app'], { cwd: __dirname });
+  });
+
+  afterEach(async function () {
+    await rimraf(`${BASE_PATH}/coverage*`);
+    await execa('git', ['clean', '-f', 'my-app'], { cwd: __dirname });
+    await execa('git', ['restore', 'my-app'], { cwd: __dirname });
+  });
```

Comment:
This prelude seems to be repeated a bunch of times, can we make it a util and replace this in each file with:

```js
setupTest(BASE_PATH);
```